script/src/verify.rs

```diff
@@ -399,8 +399,7 @@ impl<'a, DL: CellDataProvider + HeaderProvider> TransactionScriptsVerifier<'a, D
 
     /// Returns the version of the machine based on the script and the consensus rules.
     pub fn select_version(&self, script: &'a Script) -> Result<ScriptVersion, ScriptError> {
-        let proposal_window = self.consensus.tx_proposal_window();
-        let epoch_number = self.tx_env.epoch_number(proposal_window);
+        let epoch_number = self.tx_env.current_epoch_number();
```

Comment:
Better add a comment explaining why we ignore the proposal window when selecting the VM version.